modules/sdk-core/src/bitgo/inscription/inscription.ts

```diff
@@ -0,0 +1,75 @@
+import assert from 'assert';
+import { IInscription, TxInfoUnspents } from './iInscription';
+import { BitGoBase } from '../bitgoBase';
+import { IWallet } from '../wallet';
+import { inscriptions, PreparedInscriptionRevealData, TapLeafScript } from '@bitgo/utxo-ord';
+import { KeyIndices } from '../keychain';
+import { xprvToRawPrv } from '../../account-lib';
+import { HalfSignedUtxoTransaction } from '../baseCoin';
+import { inferNetworkFromCoinName } from '../utils';
+
+export class Inscription implements IInscription {
+  private readonly bitgo: BitGoBase;
+  private readonly wallet: IWallet;
+
+  constructor(bitgo: BitGoBase, wallet: IWallet) {
+    this.bitgo = bitgo;
+    this.wallet = wallet;
+  }
+
+  async prepareReveal(inscriptionData: Buffer, contentType: string): Promise<PreparedInscriptionRevealData> {
+    const user = await this.wallet.baseCoin.keychains().get({ id: this.wallet.keyIds()[KeyIndices.USER] });
+    assert(user.pub);
+    const pubkey = Buffer.from(user.pub);
+    const network = inferNetworkFromCoinName(this.wallet.coin());
+
+    return inscriptions.createInscriptionRevealData(pubkey, contentType, inscriptionData, network);
+  }
+
+  /**
+   *
+   * @param walletPassphrase
+   * @param tapLeafScript
+   * @param unsignedCommitTx
+   * @param commitTransactionUnspents
+   * @param commitAddress
+   * @param recipientAddress
+   */
+  async signAndSendReveal(
+    walletPassphrase: string,
+    tapLeafScript: TapLeafScript,
+    commitAddress: string,
+    unsignedCommitTx: string,
+    commitTransactionUnspents: TxInfoUnspents[],
+    recipientAddress: string
+  ): Promise<any> {
```

Comment:
Can we type this `any` as well?

Reply:
@OttoAllmendinger @mmcshinsky-bitgo  added a type for this. lmk if it looks good. Only thing is im not sure which fields should be optional. Used a response from submit transaction to create the type.